Configuration file merge

xenog opened the pull request:
Merge and simplify configuration files.

plaprade commented on Network/Haskoin/Wallet/Client.hs:
```diff
@@ -45,131 +55,102 @@ warningMsg = unwords
+
+getConfig :: [(Config -> Config)] -> IO Config
+getConfig fs = do
+    home <- fromMaybe (error "HOME environment not set") <$> getEnv "HOME"
```
Do we always need a HOME environment? I know you changed that in a previous pull request.

xenog commented:
I have addressed some concerns:

- The configuration file and working directory can be specified on the command line too.
- The software does best effort to determine working directory and configuration file without HOME.
- Separate database sections for MySQL and SQLite in configuration file.

plaprade commented:
ACK
